### PR TITLE
fix: make sure grammar roundtrip work for basic M2M test case

### PR DESCRIPTION
**Pull request by akphi**
Related to #165

The one problem remains with this is the fact that grammar parser adds a parameter for `src` to the property mapping lambda, which Studio excludes intentionally, @MauricioUyaguari check if we can remove this addition from the grammar parser side and so we should be good with this case

### CODE REVIEW

File: packages/legend-studio-manual-tests/src/__tests__/roundtrip-grammar/RoundtripGrammar.test.ts [hunk 1]
```diff
@@ -15,6 +15,32 @@
  */
 
 /// <reference types="jest-extended" />
+
+// NOTE: mock these methods to make sure we rule out false positive. The grammar parser for any List type field,
```
Comment (MauricioUyaguari) [inline packages/legend-studio-manual-tests/src/__tests__/roundtrip-grammar/RoundtripGrammar.test.ts:19]: perfect

File: packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_ConnectionTransformer.ts [hunk 2]
```diff
@@ -239,7 +240,8 @@ const transformJsonModelConnection = (
 ): V1_JsonModelConnection => {
   const connection = new V1_JsonModelConnection();
   connection.class = V1_transformElementReference(element.class);
-  connection.store = V1_transformElementReference(element.store);
+  // to make this consistent with grammar parser, ModelStore is a pseudo-store and will be skipped
+  connection.store = '';
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_ConnectionTransformer.ts:244]: would making store optional be problematic here?

Comment (MauricioUyaguari) [reply packages/legend-studio/src/models/protocols/pure/v1/transformation/pureGraph/from/V1_ConnectionTransformer.ts:244]: Can we leave this as undefined?

File: packages/legend-studio-manual-tests/src/__tests__/roundtrip-grammar/RoundtripGrammar.test.ts (hunk 1 above)
Comment (akphi) [reply packages/legend-studio-manual-tests/src/__tests__/roundtrip-grammar/RoundtripGrammar.test.ts:19]: 🙇‍♂️

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**